homeassistant/components/airnow/sensor.py

```diff
@@ -148,13 +148,14 @@ def __init__(
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
+
+        _device_id = f"{coordinator.latitude}-{coordinator.longitude}"
+
         self.entity_description = description
-        self._attr_unique_id = (
-            f"{coordinator.latitude}-{coordinator.longitude}-{description.key.lower()}"
-        )
+        self._attr_unique_id = f"{_device_id}-{description.key.lower()}"
         self._attr_device_info = DeviceInfo(
             entry_type=DeviceEntryType.SERVICE,
-            identifiers={(DOMAIN, self._attr_unique_id)},
+            identifiers={(DOMAIN, _device_id)},
```

Comment:
Changing the device identifier requires a migration of the old device entry properties to the new device entry so that user customizations aren't lost.

We should also add tests for the migration.

Reply:
The problem is they all had multiple devices. How do we mergre device entries.

Reply:
Pick one? Better to be correct 25 % than 0%.